pkg/volume/empty_dir/empty_dir.go

```diff
@@ -257,6 +267,36 @@ func (ed *emptyDir) setupTmpfs(dir string) error {
 	return ed.mounter.Mount("tmpfs", dir, "tmpfs", nil /* options */)
 }
 
+// setupHugepages creates a hugepage mount at the specified directory.
+func (ed *emptyDir) setupHugepages(dir string) error {
+	if ed.mounter == nil {
+		return fmt.Errorf("memory storage requested, but mounter is nil")
+	}
+	if err := ed.setupDir(dir); err != nil {
+		return err
+	}
+	// Make SetUp idempotent.
+	medium, isMnt, err := ed.mountDetector.GetMountMedium(dir)
+	if err != nil {
+		return err
+	}
+	// If the directory is a mountpoint with medium memory, there is no
+	// work to do since we are already in the desired state.
+	if isMnt && medium == mediumHugepages {
+		return nil
+	}
+	if ed.size == "" {
+		return fmt.Errorf("size is not provided")
+	}
+	if _, err := resource.ParseQuantity(ed.size); err != nil {
+		return fmt.Errorf("unsupported hugetlbfs size: %v", err)
+	}
+
+	glog.V(3).Infof("pod %v: mounting hugepages for volume %v", ed.pod.UID, ed.volName)
+	mountOptions := []string{fmt.Sprintf("size=%s", ed.size)}
```

Comment:
i don't understand why you have size.

size is bounded by sum resource requirement across containers on pod.

we need a new field specific for hugepages that aligns with pagesize. mount option.